app/src/main/kotlin/compiler/Compiler.kt

```diff
@@ -24,8 +37,9 @@ class Compiler(val diagnostics: Diagnostics) {
 
     private val lexer = Lexer(LanguageTokens.getTokens(), diagnostics)
     private val parser = Parser(LanguageGrammar.getGrammar(), diagnostics)
+    private val covering = DynamicCoveringBuilder(InstructionSet.getInstructionSet())
 
-    fun process(input: Reader, output: Writer): Boolean {
+    fun process(input: Reader, output: PrintWriter): Boolean {
```

Comment:
It could have remained as `Writer` (for symmetry with `Reader`), and then it could be wrapped into `PrintWriter`.